go/libraries/doltcore/sqle/enginetest/dolt_harness.go

```diff
@@ -219,13 +241,15 @@ func (d *DoltHarness) NewReadOnlyDatabases(names ...string) (dbs []sql.ReadOnlyD
 }
 
 func (d *DoltHarness) NewDatabaseProvider(dbs ...sql.Database) sql.MutableDatabaseProvider {
-	if d.env == nil {
-		d.env = dtestutils.CreateTestEnv()
+	// Because NewDatabases is called before NewDatabaseProvider, we need to re-add the
+	// test DoltEnvs created in NewDatabases to the new MultiRepo created here.
```

Comment:
or make the multienv here? or add the envs when we make the database? i know you are constrained by the existing interfaces, but it seems like one of these methods should be doing more work. Like should `NewDatabaseProvider` even take databases as an argument if they are already in `d.databases`? My guess is that either 1) new databases should not persist the databases, and everything should happen here, or 2) NewDatabases does the multienv setup work and NewDatabaseProvider uses `d.databases` instead of the passed dbs. Leaning towards (2), zach will have better ideas.